### DISCUSSION TITLE
JS 논리연산자

### DISCUSSION THREAD

**Discussion by WooGie911** [wanted-9th-3team/frontend-study#62] [프론트 기술 질문 공간]
다음 결과를 예측해보세요

1. `alert( null || 2 || undefined );`

2. `alert( alert(1) || 2 || alert(3) );`

3. `alert( 1 && null && 2 );`

4. `alert( null || 2 && 3 || 4 );`

**Comment by michoball** [comment]
1. 위 alert 안에 참인 값은 2 이므로 2로 alert 창이 뜹니다. 
2. alert 안 alert(1)을 먼저 실행합니다. alert으로 1이 뜨고 undefined를 반환합니다. 그리고 2을 true 값으로 평가해서 alert(2)를 실행하고 끝납니다. 
3. 1이 참이면 다음 값을 실행합니다. 다음 값은 null 인데 null 을 false 값이므로 null에서 멈추고 alert(null) 을 하게 됩니다. 
4.  처음에 있는 null이 false 값이므로 다음 판별문으로 갑니다.  2 가 true면 3을 사용한다는 연산이므로  식이 맞기 때문에  alert 는 3을 띄웁니다.

**Comment by WooGie911** [comment]
1. null || 2 || undefined 가 2 이므로 2로 alert 창이 뜹니다.
2. alert(1) 이 undefined 이므로 alert(1) || 2 || alert(3) === undefined  || 2 || undefined 이고, 즉 undefined  || 2 || undefined 가 2 이므로  alert(2) 의 결과가 나옵니다. 하지만 alert() 안에서  alert(1)이 첫번째 truthy를 반환했기에 실행이 멈추고 바깥 alert() 가 실행 되어 alert(2)가 출력 됩니다.
3. 1 && null && 2 === null 이므로 null로 alert 창이 뜹니다.
4. null || 2 && 3 || 4 === 3 이므로 3으로 alert 창이 뜹니다.
